utils/utils.php

```diff
@@ -61,10 +61,26 @@ function tlms_isApiKey($apiKey){
 	}
 }
 
+if(!function_exists('tlms_parseDate')){
+	function tlms_parseDate($format, $date){
+		$isPM = (stripos($date, 'PM') !== false);
+		$date = str_replace(array('AM', 'PM'), '', $date);
+
+		$date = DateTime::createFromFormat(trim($format), trim($date));
+
+		if ($isPM){
+			$date->modify('+12 hours');
+		}
+
+		return $date;
+	}
+}
+
 if(!function_exists('tlms_getDateFormat')){
 	function tlms_getDateFormat($no_sec = false){
+		// TODO: Store the site info in the database instead of hitting the API everytime we want to get it.
 		$site_info = tlms_getTalentLMSSiteInfo();
-		$date_format = $site_info['date_format'];
+		$date_format = $site_info instanceof Exception ? '' : $site_info['date_format'];
```

Comment:
I think that we should probably throw the exception instead of returning it but we are doing as you said in a lot of methods. So lets just keep it that way for now.